lib/test_prof/recipes/rspec/let_it_be.rb

```diff
@@ -114,6 +113,89 @@ def define_let_it_be_methods(identifier, **modifiers)
 
       let(identifier, &let_accessor)
     end
+
+    def define_freezing_hooks
+      # Prevent hooks from being defined several times
+      return if instance_variable_get(:"#{PREFIX}hooks_defined")
+
+      before(:all) do
+        let_it_be_objects = instance_variable_get(:"#{TestProf::LetItBe::PREFIX}let_it_be_objects")
+
+        let_it_be_objects.each { |object| Freezer.deep_freeze(object) }
+      end
+
+      prepend_after do |example|
+        if example.exception&.message&.match?(/can't modify frozen Hash/)
+          example.exception.message << FROZEN_HASH_HINT
+        end
+      end
+
+      instance_variable_set(:"#{PREFIX}hooks_defined", true)
+    end
+
+    # Exception needs to be handled both here and in `handle_frozen_hash_error`
+    # because if it is raised in before_all it isn't caught in `after` block and
+    # if it's inside the example it isn't raised so it has to be handled in `after`.
+    def build_let_it_be_initializer(identifier, freeze, &block)
+      proc do
+        begin
+          record = instance_exec(&block)
+          if freeze
+            # FIXME: When models and their associations are defined with different
+            # options, e.g. `reload: true` and without it, the ones that are not
+            # supposed to be frozen will still be frozen here. Add those with
+            # `reload: true`/`refind: true`/`freeze: false` to ignore list.
+            let_it_be_objects = instance_variable_get(:"#{TestProf::LetItBe::PREFIX}let_it_be_objects")
+            let_it_be_objects ||= instance_variable_set(:"#{TestProf::LetItBe::PREFIX}let_it_be_objects", [])
+            let_it_be_objects << record
+          end
+
+          instance_variable_set(:"#{TestProf::LetItBe::PREFIX}#{identifier}", record)
+        rescue => e
+          e.message << FROZEN_HASH_HINT if e.message.match?(/can't modify frozen Hash/)
+          raise e
+        end
+      end
+    end
+
+    module Freezer
+      # Rerucsively freezes the object to detect modifications.
+      def self.deep_freeze(record)
+        return if record.frozen?
+
+        record.freeze
+
+        return record.each { |rec| deep_freeze(rec) } if record.respond_to?(:each)
+
+        # Freeze associations as well.
+        #
+        # NOTE: `reload` statements in test or production code will cause
+        # a `FrozenError`. In case the use of `reload` cannot be avoided, use
+        # `reload: true` in `let_it_be` declaration.
+        return unless record.is_a?(::ActiveRecord::Base)
```

Comment:
We need to check for `defined?(ActiveRecord::Base)` first. Not everyone uses ActiveRecord)

Reply:
Makes sense. I'll add a guard statement before this one to check if `AR::B` is defined. Everything below this relates to AR.